resources/tenant-onboarding-app.yaml

```diff
@@ -431,6 +436,11 @@ Resources:
                   - s3:GetObjectVersion
                 Resource:
                   - !Sub arn:${AWS::Partition}:s3:::{{resolve:ssm:/saas-boost/${Environment}/RESOURCES_BUCKET}}/tenants/${TenantId}/*
+              - Effect: Allow
+                Action:
+                  - s3:*
+                Resource:
+                  - !Sub arn:${AWS::Partition}:s3:::${S3ExtensionBucket}/${ServiceResourceName}/${TenantId}/*
```

Comment:
This is too permissive. There are bucket level actions we don't want the SaaS application to be able to modify. SaaS Boost owns the provisioning and configuration of this bucket.

```yaml
- Effect: Allow
  Action:
    - s3:ListBucket
    - s3:ListBucketVersions
  Resource:
    - !Sub arn:${AWS::Partition}:s3:::${S3ExtensionBucket}
  Condition:
    StringLike:
      s3:prefix:
        - !Sub ${ServiceResourceName}/${TenantId}/*
- Effect: Allow
  Action:
    - s3:GetBucket*
    - s3:GetAccelerateConfiguration
    - s3:GetAnalyticsConfiguration
    - s3:GetEncryptionConfiguration
    - s3:GetIntelligentTieringConfiguration
    - s3:GetInventoryConfiguration
    - s3:GetLifecycleConfiguration
    - s3:GetMetricsConfiguration
  Resource:
    - !Sub arn:${AWS::Partition}:s3:::${S3ExtensionBucket}
- Effect: Allow
  Action:
    - s3:GetObject*
    - s3:PutObject*
    - s3:DeleteObject*
    - s3:AbortMultipartUpload
    - s3:ListMultipartUploadParts
  Resource:
    - !Sub arn:${AWS::Partition}:s3:::${S3ExtensionBucket}/${ServiceResourceName}/${TenantId}/*
```